### ISSUE TITLE
✍️ Contribution period: juliet owanku

### ISSUE THREAD

**Issue by julietowah** [ersilia-os/ersilia#849]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [X] Install the Ersilia Model Hub and test the simplest model
- [X] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Install and run an ML model

- [x] Select a model from the suggested list
- [x] Install the model in your system
- [x] Run predictions for the EML
- [x] Compare results with the Ersilia Model Hub implementation!
- [x] Install and run Docker!

### Week 3 - Propose new models

- [x] Suggest a new model and document it (1)
- [x] Suggest a new model and document it (2)
- [x] Suggest a new model and document it (3)

### Week 4 - Prepare your final application

- [x] Submit the final application in the Outreachy website

**Comment by julietowah** [comment]
**Join the communication channels**
i was able to join the communication channel on slack and am so excited to meet few people who has been helping me to set up my environment.

**Comment by julietowah** [comment]
**Install the Ersilia Model Hub and test the simplest model**
i have successfully installed the  Ersilia model Hub and tested it using

ersilia -v fetch eos3b5e
ersilia serve eos3b5e
ersilia -v api run -i "CCCC"

and my output is:

{
    "input": {
        "key": "IJDNQMDRQITEOD-UHFFFAOYSA-N",
        "input": "CCCC",
        "text": "CCCC"
    },
    "output": {
        "mw": 58.123999999999995
    }
}

**Comment by julietowah** [comment]
**My motivation statement:**
Hello everyone my name is juliet Owanku am from Nigeria. am still growing my skill to becoming a software engineer and i have been opportune to do some projects using Html, css, javascript, and python.

One thing that kept me motivated in pursuing this career was my urge to joining in bringing solution using tech. i have been making some progress in trying to find a problem and creating solution for it . 

i was so happy upon coming across Ersilia project because its just Everything i wanted (to become part of a solution system) and i am so happy to contribute to this project and hopefully having a great impact to the society in the nearest future.

For me coming from this part of the world i know that this project will be of a great relief in the health sector and i look forward to leaning and improving my skill in programming generally.
thank you

**Comment by julietowah** [comment]
*Select a model from the suggested list*
i choose **SARS-CoV2 activity (Image Mol)**  because of these reasons:

after the global shot down to cob covid-19, the virus has mutilated and formed more mutant virus which made it difficult to keep track but with this discovery an AI ImageMol which is trained on molecular images to predict molecular targets of candidate compounds is created and this will help in predicting potential drugs only by identifying the molecular properties of the virus.
this can be a big a very big breakthrough in medical science

**Comment by julietowah** [comment]
hi @DhanshreeA please i would love your help on the imagmol pretrain am kinda lost and i would love to get back on track

**Comment by leilayesufu** [comment]
Hi Juliet, Where are you lost?
Have you installed the model on your system?

**Comment by julietowah** [comment]
i created new environment using 
```
conda create -n imagemol python=3.7.3
```
then avtivated it with
```
conda activate imagemol
```
dounloaded the packeges for pytorch, torch-cluster torch-scatter torch-sparse torch-spline-conv
```
conda install -c rdkit rdkit
pip install https://download.pytorch.org/whl/cu101/torch-1.4.0-cp37-cp37m-linux_x86_64.whl
pip install https://download.pytorch.org/whl/cu101/torchvision-0.5.0-cp37-cp37m-linux_x86_64.whl
pip install torch-cluster torch-scatter torch-sparse torch-spline-conv -f https://pytorch-geometric.com/whl/torch-1.4.0%2Bcu101.html
```
then i was able to download the pretrain dataset and saved it in ./datasets/pretraining/data/
preprocess the dataset

then got lost at 
start to pretrain](https://github.com/HongxinXiang/ImageMol#2-start-to-pretrain)
i just need some explanations and directions 
thank you

**Comment by leilayesufu** [comment]
Hi, i just went through the model and i don't think you're to pretrain but rather make predictions using a pre-trained model.

**Comment by leilayesufu** [comment]
This should be the model you're to use. "SARS-CoV2 activity (Image Mol)"

If you've installed the neccessary packages, you can skip pre-training and go to finetuning in the github repository

**Comment by julietowah** [comment]
ok thank you so much @leilayesufu

**Comment by julietowah** [comment]
###**Install the model in your system**

i first cloned the repo with
```
git clone --recursive https://github.com/ncats/ncats-adme.git
```
**Installing required software**
as i have previously installed miniconda in my system
i just had to create the environment in my terminal
```
ADME_RLM/server
```
created the environment using
```
conda env create --prefix ./env -f environment.yml
```
i activated the environment with
```
conda activate ./env
```

**Comment by julietowah** [comment]
**Running the application**
i run the code with
```
python app.py
```

**Comment by julietowah** [comment]
## **Run predictions for the EML**

**Comment by julietowah** [comment]
**Ran the prediction again using selecting PAMPA Permeability (pH 7.4) and PAMPA Permeability (pH 5.0)**

![image](https://github.com/ersilia-os/ersilia/assets/104780614/975f300b-3b40-41de-85b1-3da5bd4b8102)

![image](https://github.com/ersilia-os/ersilia/assets/104780614/8fc29f34-28ad-4a13-98e4-3192bed5d58a)

PAMPA Permeability (pH 7.4)  and PAMPA Permeability (pH 5.0) which means Parallel Artificial Membrane Permeability Assay.Permeability measures the ability of a substance (often a drug or chemical compound) to cross a model membrane under specific conditions. in this case measurement is performed at a pH of 7.4 and 5.0 respectively.

from the above illustration  High permeability indicates that the compounds are expected to pass through biological membranes more readily, which can be important in drug absorption. Low or moderate permeability suggests a slower or controlled rate of permeation. so in the first row of the PAMPA Permeability (pH 7.4) table it has 
**low or moderate permeability** with the probability of 1(1.0) and the second row has 
**high permeability with 0.9 proberbility**
in PAMPA Permeability (pH 5.0) the first ro has low permiability with proberbility of 1(0.9) and the second row has moderate or high permeability with proberbility of 0(1.0)

**Comment by julietowah** [comment]
## **Compare results with the Ersilia Model Hub implementation!**

I opend the [Ersilia Model Hub](https://www.ersilia.io/model-hub) link and clicked on the tab Microsomal stability. which has the  Human Liver Microsomal Stability and Rat liver microsomal stability. I clicked on the Human liver microsomal stability and was redirected to the Ersilia GitHub [repo](https://github.com/ersilia-os/eos31ve) of the model. went through the README.md in the repo and clicked on the [DockerHub](https://hub.docker.com/r/ersiliaos/eos31ve) link.

The Human liver microsomal stability model code is eos31ve.

I opened a terminal and ran the code below to fetch the model.

```ersilia -v fetch eos31ve```

and i got this output

**Output**

``
09:09:44 | DEBUG    | Getting schema for API run...
09:09:44 | DEBUG    | No annotated metadata could be retrieved
09:09:44 | DEBUG    | No annotated metadata could be retrieved
09:09:44 | DEBUG    | Latest meta: {'hlm_proba1': ['hlm_proba1']}
09:09:44 | DEBUG    | hlm_proba1 : {'type': 'numeric'}
09:09:44 | DEBUG    | Meta k: ['hlm_proba1']
09:09:44 | DEBUG    | Schema: {'input': {'key': {'type': 'string'}, 'input': {'type': 'string'}, 'text': {'type': 'string'}}, 'output': {'hlm_proba1': {'type': 'numeric', 'meta': ['hlm_proba1']}}}
09:09:44 | DEBUG    | Done with the schema!
09:09:44 | DEBUG    | This is the schema {'input': {'key': {'type': 'string'}, 'input': {'type': 'string'}, 'text': {'type': 'string'}}, 'output': {'hlm_proba1': {'type': 'numeric', 'meta': ['hlm_proba1']}}}
09:09:44 | DEBUG    | API schema saved at /home/juliet/eos/dest/eos31ve/api_schema.json
09:09:47 | DEBUG    | Fetching eos31ve done in time: 0:05:13.409397s
09:09:47 | INFO     | Fetching eos31ve done successfully: 0:05:13.409397
👍 Model eos31ve fetched successfully!
``

i then served the model useing

``` ersilia serve eos31ve ```

and the output is 

```
🚀 Serving model eos31ve: ncats-hlm

   URL: http://127.0.0.1:52765
   PID: 7917
   SRV: conda

👉 To run model:
   - run

💁 Information:
   - info
```

**Comment by julietowah** [comment]
## **Install and run Docker!**

i installed docker using this code:
``` sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin ```
and tested it with:
``` sudo docker run hello-world ```

**Comment by julietowah** [comment]
## **1st Model Suggestion**

**KGNN: Knowledge Graph Neural Network for Drug-Drug Interaction Prediction**

Publication: [https://xuanlin1991.github.io/files/publications/ijcai20.pdf](url)

**Comment by DhanshreeA** [comment]
Hi @julietowah 

Thank you for the very detailed updates, and apologies for responding late. 

If you find yourself finished with Week 3's tasks sooner and you still have some time remaining, I would recommend a bonus task (Please note, not completing this will not count towards your application, so no pressure there.)
Can you extract the code required to run one of the NCATS Models (either one of RLM or HLCS) into a simple python script and run it against the EML? So in short, it should do the following:
- Take SMILE inputs and preprocess them as per the original code
- Load the model from a model path
- Run predictions 

The NCATS repo provides a full fledged server with a lot of extra functionality that models within the Ersilia Model Hub do not need. When we implement models within the hub, we try to keep the bare minimum required to process inputs as needed, load a model from a local file path, and make predictions. This generally involves extracting only the essential parts from the original codebase.

**Comment by DhanshreeA** [comment]
> ## **1st Model Suggestion**
> 
> **KGNN: Knowledge Graph Neural Network for Drug-Drug Interaction Prediction**
> 
> Publication: [KGNN](https://xuanlin1991.github.io/files/publications/ijcai20.pdf) Source Code (TensorFlow): [sourcecode](https://github.com/xzenglab/KGNN) Dataset: [V5.1.4](https://www.drugbank.ca/releases/5-1-4)
> 
> **About this model** Drug-Drug Interaction (DDI) prediction is a critical area within pharmacology and drug discovery that aims to anticipate and assess potential interactions between different drugs when taken simultaneously. Understanding these interactions is vital for patient safety and optimizing therapeutic outcomes, as some drug combinations can lead to adverse effects, reduced efficacy, or unexpected outcomes. DDI prediction plays a pivotal role in drug safety and personalized medicine, guiding clinicians and patients in making informed decisions regarding medication usage and combinations.
> 
> **Why this model matters to Ersilia**
> 
> Drug-Drug Interaction (DDI) prediction is crucial for Ersilia as it aligns with their mission of providing cutting-edge computational tools and models to researchers, pharmaceutical companies, and healthcare professionals in the field of drug discovery and development. Here's how DDI prediction is important to Ersilia:
> 
> Safety and Efficacy: Predicting potential drug-drug interactions is fundamental to ensure the safety and efficacy of drug therapies. By offering accurate DDI prediction models, Ersilia helps in identifying possible interactions that could lead to adverse effects or reduced drug efficacy, enabling better-informed decisions.
> 
> Patient-Centric Approach: Ersilia's models contribute to a patient-centric approach to healthcare. Predicting DDIs helps personalize treatment plans, ensuring that prescribed drugs are compatible and safe for individual patients, considering their unique health conditions and existing medication regimens.
> 
> Enhanced Drug Development: Drug developers can utilize DDI prediction models during the drug development process to assess potential interactions early on. This can streamline the development pipeline, reduce costs, and improve the chances of bringing successful, safe drugs to market.
> 
> Data-Driven Insights: Ersilia's DDI prediction models leverage data and advanced computational techniques. By providing data-driven insights into potential DDIs, Ersilia empowers researchers and practitioners with valuable information to optimize drug combinations and mitigate risks.
> 
> Support for Healthcare Professionals: Healthcare professionals can benefit from Ersilia's DDI prediction models to make informed decisions about drug prescribing. This support is critical in clinical settings, where physicians need quick and accurate information about potential interactions to provide the best care for their patients.
> 
> Research and Innovation: Ersilia's involvement in DDI prediction showcases their commitment to research and innovation. Continuously improving and expanding DDI prediction models contributes to the advancement of pharmacology and fosters innovation in drug discovery and healthcare practices.
> 
> **Code Implementation**
> 
> To run the code, you need the following dependencies:
> 
> Python == 3.6.6 Keras == 2.3.0 Tensorflow == 1.13.1 scikit-learn == 0.22
> 
> The TensorFlow-GPU library has progressed beyond version 1.13.1, with the most recent releases belonging to the 2.x series. Therefore, it's important to update the model to ensure compatibility and optimal performance on the latest versions.

Hi @julietowah very interesting paper! Thank you for recommending this. Unfortunately the inputs within this model do not seem to be conformant with what Ersilia Model Hub expects at the time (ie SMILES strings). I may have missed something within the code, please feel free to correct me.

**Comment by julietowah** [comment]
## **1st Model Suggestion**
decided to use this as my 1st suggestion
ADMETlab 2.0: an integrated online platform for accurate and comprehensive predictions of ADMET properties
Publication: [ADMET](https://academic.oup.com/nar/article/49/W1/W5/6249611?searchresult=1)
Source Code: [sourcecode MGA](https://github.com/wzxxxx/MGA/tree/main)
model name : MGA

**about this model**:

ADMET (absorption, distribution, metabolism, excretion, and toxicity) properties are crucial in drug development to ensure a balance between potency, pharmacokinetics, and safety of drug candidates. Traditionally, ADMET evaluation followed potency determination, often leading to late-stage adverse effects discovery. The role of ADMET evaluation has become pivotal, contributing to nearly 50% of drug development attrition. In silico prediction models and web tools have emerged to efficiently predict ADMET properties, aiding early-stage compound screening and lead optimization.

**Why this model matters to Ersilia**

 ADMETlab 2.0 aligns with Erisia's goals in drug discovery and development by offering a powerful tool for predicting crucial properties of drug candidates, ultimately contributing to more effective and efficient drug development processes.

**Code Implementation**
**requirements：**
python 3.6
anaconda
dgl 0.4.3
xgboost
rdkit
pytorch
sklearn
ADMETlab 2.0 is a web application built using the Python web framework Django and deployed on an Ubuntu Linux system hosted on Aliyun's elastic compute service. The web access is facilitated through the Nginx web server,The prediction models are implemented using Python with deep learning packages PyTorch and DGL, as well as the RDKit package for cheminformatics support.

**Comment by julietowah** [comment]
## **2nd Model Suggestion**
Deep learning to generate in silico chemical property libraries and candidate
molecules for small molecule identification in complex samples

Publication: [Deep learning](https://arxiv.org/pdf/1905.08411v1.pdf)
Source Code: [sourcecode ](https://github.com/pnnl/darkchem)
model name : darkchem

**About this model:**
DarkChem is a powerful framework that advances the field of small molecule characterization, particularly in metabolomics. Its ability to predict properties directly from molecular structures, focus on metabolomics-related properties, and adapt to different property prediction tasks positions it as a valuable tool for researchers working with complex mixtures and experimental data in various scientific disciplines.

**why this model matters to Erisilia**

DarkChem's ability to identify and generate small molecule structures in complex mixtures and can enhance the understanding of metabolite roles and interactions within biological processes which can aid in identifying potential drug candidates and their properties, streamlining the drug design and optimization process

**implementation:**

DarkChem was written in Python(version 3.6) and uses Keras with Tensorflow and from my observation i think it is ready to use

**Comment by julietowah** [comment]
hi @DhanshreeA  do you think this https://github.com/uzh-dqbm-cmi/side-effects/  is okey for suggestion

**Comment by julietowah** [comment]
## **3rd Model Suggestion**

Inductive transfer learning for molecular activity prediction: Next-Gen QSAR Models with MolPMoFiT

Publication: [MolPMoFiT](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00430-x)
Source Code: [sourcecode](https://github.com/XinhaoLi74/MolPMoFiT)
model name : MolPMoFiT

**About this model:**

MolPMoFiT is a powerful approach for predicting molecular properties accurately, even with smaller and challenging datasets. It combines self-supervised pre-training on a large dataset of unlabeled compounds with task-specific fine-tuning. This approach outperforms other machine learning methods on benchmark datasets, making it valuable for diverse chemical prediction tasks.
MolPMoFiT aids in drug discovery by improving the accuracy and reliability of predicting molecular properties, which is crucial in the early stages of drug development. It can effectively handle smaller and challenging datasets, making it a valuable tool for optimizing drug candidates. This approach streamlines the process of identifying compounds with desired properties, ultimately speeding up drug discovery and reducing costs. Additionally, by utilizing self-supervised learning on a vast dataset of unlabeled compounds, MolPMoFiT harnesses the wealth of publicly-available chemical information to enhance prediction models, facilitating more informed decision-making in drug development.

**How dose this model matter to Erisilia:**

MolPMoFiT's capabilities are highly relevant to Ersilia's mission, which involves advancing research and innovation in various scientific fields, including drug discovery and cheminformatics. Here's how MolPMoFiT can contribute to Ersilia's mission:

Accelerating Drug Discovery: MolPMoFiT enhances the drug discovery process by improving the accuracy of predicting molecular properties. This acceleration can lead to the discovery of new drug candidates more rapidly, aligning with Ersilia's goal of advancing research and innovation.

Cost Reduction: By increasing the efficiency and reliability of predicting molecular properties, MolPMoFiT can help reduce the costs associated with drug development. This cost-saving aspect is essential for optimizing resource allocation in scientific endeavors.

Leveraging Public Datasets: MolPMoFiT utilizes a vast dataset of unlabeled compounds from public sources. This approach aligns with Ersilia's mission of promoting the open sharing of scientific knowledge and resources for the benefit of the research community.

Promoting Data-Driven Research: Ersilia encourages data-driven approaches in scientific research. MolPMoFiT is a prime example of harnessing data to improve prediction models, contributing to more informed and data-driven decision-making in scientific endeavors.

In summary, MolPMoFiT's capabilities directly align with Ersilia's mission by advancing research, promoting open data sharing, and supporting cost-effective and data-driven scientific investigations. It can be a valuable asset for researchers and organizations committed to Ersilia's goals.

**code implementation:**

**Comment by julietowah** [comment]
@DhanshreeA  about writing the python script 

i have some errors i am running into which i would love fix first though i will continue to work on it and hopefully be able to solve it 
i really appreciate the extra project i would really love to get the final script

**Comment by GemmaTuron** [comment]
Hello,

Thanks for your work during the Outreachy contribution period, we hope you enjoyed it! We will now close this issue while we work on the selection of interns. Thanks again!
